src/parser/whamm_parser.rs

```diff
@@ -169,13 +165,21 @@ pub fn process_pair(whamm: &mut Whamm, script_count: usize, pair: Pair<Rule>, er
                 None => (None, None)
             };
 
+            let this_body: Option<Vec<Statement>> = this_body.map(|b| {
+                let mut stmts = vec![];
+                for stmt in b {
+                    match stmt {
+                        Ok(s) => stmts.push(s),
+                        Err(errors) => err.add_errors(errors)
+                    }
+                }
+                stmts
+            });
```

Comment:
Are you sure this works? Tracing through the logic, I think you're missing a `.collect()` or something to actually pull all the values together to have a complete vec of statements.

Reply:
I think I'm safe because I pushed them one by one?

Reply:
Ahhh, it's mapping to the unwrapped `Some` contents and then iterating in the `for` loop. That makes sense.